### PR TITLE
LPS-100144 - SF

### PR DESCRIPTION
**Pull request by carloslancha**

### CODE REVIEW

File: modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/categorization/tags/EditTags.es.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+	const [selectAll, setSelectAll] = useState();
+	const [folderId, setFolderId] = useState();
+	const [showModal, setShowModal] = useState();
+
+	const handleOnClose = () => {
+		setShowModal(false);
+	};
+
+	if (!Liferay.component(props.componentId)) {
+		Liferay.component(props.componentId, {
```
Comment (jbalsas) [inline modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/categorization/tags/EditTags.es.js]: This is an interesting pattern that we need to further explore to understand how to make it available

File: modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/view.jsp [hunk 2]
```diff
@@ -291,14 +291,32 @@ String navigation = ParamUtil.getString(request, "navigation");
 		<%
 		String pathModule = PortalUtil.getPathModule();
 
+		long groupIds[] = PortalUtil.getCurrentAndAncestorSiteGroupIds(scopeGroupId);
```
Comment (jbalsas) [inline modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/view.jsp]: We should send this to the 🍋 team to review 👍

File: modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/view.jsp [hunk 3]
```diff
@@ -307,18 +325,11 @@ String navigation = ParamUtil.getString(request, "navigation");
...
-			context="<%= tagsContext %>"
-			module="document_library/js/categorization/EditTags.es"
-		/>
-
 		<liferay-frontend:component
 			componentId='<%= liferayPortletResponse.getNamespace() + "EditCategoriesComponent" %>'
 			containerId='<%= "#" + liferayPortletResponse.getNamespace() + "documentLibraryModal" %>'
 			context="<%= categoriesContext %>"
-			module="document_library/js/categorization/EditCategories.es"
+			module="document_library/js/categorization/soy/EditCategories.es"
```
Comment (jbalsas) [inline modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/view.jsp]: I assume we're moving `soy/EditCategories` at the same time as `EditTags`? We might want to just split that into a different LPS/PR. Just move legacy code out. Then send our new changes, so the diff is easier to read.

Comment (carloslancha) [reply modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/view.jsp]: I think we shouldn't move `EditCategories` on this pr, only deal with EditTags and AssetTagsSelector. @julien what do you think?

Comment (julien) [reply modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/view.jsp]: I agree with you too @jbalsas one PR to "move" the previous components and another to introduce the new ones.
You're right @carloslancha I shouldn't have anticipated that.

File: modules/apps/document-library/document-library-web/package.json [hunk 4]
```diff
@@ -1,13 +1,25 @@
...
+		"@clayui/form": "3.0.0-milestone.3",
+		"@clayui/icon": "3.0.0-milestone.3",
+		"@clayui/loading-indicator": "3.0.0-milestone.3",
+		"@clayui/modal": "3.0.0-milestone.3",
+		"asset-taglib": "*",
 		"clay-button": "2.18.0",
 		"clay-multi-select": "2.18.0",
 		"clay-radio": "2.18.0",
 		"clipboard": "2.0.4",
+		"frontend-js-react-web": "*",
```
Comment (julien) [inline modules/apps/document-library/document-library-web/package.json]: Is this dependency still needed, now that we're using that taglib? 
I added it there when I was still calling `render` manually.

Comment (jbalsas) [reply modules/apps/document-library/document-library-web/package.json]: It's not

File: modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/categorization/tags/EditTagsModal.es.js [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+
+	const fetchResult = useRef();
+
+	useEffect(() => {
+		fetchResult.current = fetchSelectedItems();
+
+		return () => {
+			fetchResult.current = null;
+		};
+	}, [fetchSelectedItems]);
```
Comment (julien) [inline modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/categorization/tags/EditTagsModal.es.js]: @wincent, Do you remember that when we first talked about this `useEffect` hook, the "dependency" array (I'm not sure it's called like that) was just an empty array?

We had a warning from React when running the lint task, so @carloslancha and I decided it would be a good idea to change it to this, tell us what you think when you can. Thanks.

File: modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/categorization/tags/EditTagsModal.es.js [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+		return Promise.all([
+			fetchTags(urlTags, 'POST', selection),
+			fetchTags(urlSelection, 'POST', selection)
+		]).then(([responseTags, responseSelection]) => {
+			if (responseTags && responseSelection) {
+				const selectedItems = (responseTags.items || []).map(
+					item => item.name
+				);
+
+				if (fetchResult.current) {
```
Comment (wincent) [inline modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/categorization/tags/EditTagsModal.es.js]: I think this is obscuring your intent here: what you mean is "if I am still mounted, update the UI", but what you've written read as "if I kicked off a fetch, update the UI". A `useMounted` hook would make this clearer, or even an inline `useRef()` + `useEffect()`; ie: something like:

```javascript
const isMounted = useRef(true);

useEffect(() => {
    isMounted.current = true;

    return () => {
       isMounted.current = false;
    };
});
```

and later on in your `then()`:

```javascript
if (isMounted.current) {
    // setState calls...
}
```

File: modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/categorization/tags/EditTagsModal.es.js [hunk 7]
```diff
@@ -0,0 +1,252 @@
...
+
+		const addedLabels = !append
+			? selectedItems
+			: selectedItems.filter(
+					item => initialSelectedItems.indexOf(item) === -1
+			  );
+
+		const removedLabels = initialSelectedItems.filter(
+			item => selectedItems.indexOf(item) === -1
+		);
```
Comment (wincent) [inline modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/categorization/tags/EditTagsModal.es.js]: These have the same quadratic behavior I noted above.

File: modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/categorization/tags/EditTagsModal.es.js [hunk 8]
```diff
@@ -0,0 +1,252 @@
...
+	const [loading, setLoading] = useState(true);
+	const [multiple, setMultiple] = useState(false);
+	const [initialSelectedItems, setInitialSelectedItems] = useState([]);
+	const [selectedItems, setSelectedItems] = useState([]);
+	const [selectedRadioGroupValue, setSelectedRadioGroupValue] = useState(
+		'add'
+	);
+
+	const {observer, onClose} = useModal({
+		onClose: onModalClose
```
[the remark this answers is not in the document]
Comment (julien) [reply modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/categorization/tags/EditTagsModal.es.js]: If I remember correctly it used to be called `onClose`, I think we just wanted to make it clear that this is about closing the modal, but since *this* (no pun intended) is the modal component it might be a bit redundant

File: modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_tags_selector/AssetTagsSelector.es.js [hunk 9]
```diff
@@ -79,7 +78,7 @@ function AssetTagsSelector({
 		if (innerInputValue) {
 			refetch();
 		}
-	}, [innerInputValue, refetch]);
+	}, [innerInputValue]);
```
Comment (carloslancha) [inline modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_tags_selector/AssetTagsSelector.es.js]: I remove refetch as a dependency here because it was causing an infinite petitions loop

File: modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/categorization/tags/EditTagsModal.es.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (carloslancha) [reply modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/categorization/tags/EditTagsModal.es.js]: I created a new `state` called `bulkSelection` and also added this, replacing previous `useEffect`:
```
useEffect(() => {
	setBulkSelection(getBulkSelection());
}, [fileEntries, folderId, repositoryId, selectAll]);

useEffect(() => {
	fetchSelectedItems();
}, [urlTags, urlSelection, bulkSelection]);

const isMounted = useRef(true);

useEffect(() => {
	isMounted.current = true;

	return () => {
		isMounted.current = false;
	};
});
```

@julien Let me know if I missed something please

File: modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/categorization/tags/EditTagsModal.es.js (hunk 6 above)
Comment (carloslancha) [reply modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/categorization/tags/EditTagsModal.es.js]: Changed as explained in previous comment

File: modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/categorization/soy/EditTags.soy [hunk 10]
```diff
@@ -95,7 +95,7 @@
 
 	<p>{$description}</p>
 
-	{if $multiple}
+	{if not $multiple}
```
Comment (jbalsas) [inline modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/categorization/soy/EditTags.soy]: Is this an intentional change?

File: modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_tags_selector/page.jsp [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+ */
+--%>
+
+<%@ include file="/asset_tags_selector/init.jsp" %>
+
+<%
+Map<String, Object> data = (Map<String, Object>)request.getAttribute("liferay-asset:asset-tags-selector:data");
+%>
+
+<react:component
```
Comment (jbalsas) [inline modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_tags_selector/page.jsp]: I think we're missing some good ol' SSR here :)

Comment (jbalsas) [reply modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_tags_selector/page.jsp]: Additionally, unwrapped as this is, it will be mounting on the parent element and we don't know what that might be... @wincent, this is a side effect we already observed. When `react:component` is not wrapped, the results can be devastating 😂

File: modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/categorization/soy/EditTags.soy (hunk 10 above)
Comment (carloslancha) [reply modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/categorization/soy/EditTags.soy]: Don't think so, @julien ?

Comment (julien) [reply modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/categorization/soy/EditTags.soy]: @carloslancha, do you remember we looked at this and saw that the compiled soy output looked like this (it really confused me)


EditTags soy js

Comment (julien) [reply modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/categorization/soy/EditTags.soy]: @carlos, @jbalsas, apparenlty I really missed something when I changed that template from `.soy` to  `jsx`, see [here](https://github.com/liferay/liferay-portal/blob/f56de07a75a0d3f0c5f919bd175ca9e12195e9e7/modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/categorization/EditTags.soy#L98).

Comment (carloslancha) [reply modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/categorization/soy/EditTags.soy]: @julien the thing was that was compiled into `if not $multiple`. Weird. Anyway, we can remove that change, right?

File: modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_tags_selector/page.jsp (hunk 11 above)
Comment (wincent) [reply modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_tags_selector/page.jsp]: Might have to add a lint for this. 😁

Comment (jbalsas) [reply modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_tags_selector/page.jsp]: Yes! A JSP linter!! 1000 times yes! Life keeps rewarding us with fascinating new stuff to do! 🙏

Comment (carloslancha) [reply modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_tags_selector/page.jsp]: @jbalsas @wincent added markup to jsp

File: modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/view.jsp (hunk 3 above)
Comment (carloslancha) [reply modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/view.jsp]: Reverted that changed

File: modules/apps/asset/asset-taglib/.babelrc [hunk 12]
```diff
@@ -0,0 +1,5 @@
+{
+    "presets": [
+        "@babel/preset-react"
```
Comment (jbalsas) [inline modules/apps/asset/asset-taglib/.babelrc]: Do we still need this, @wincent? Will it conflict with other presets if we enable it by default?

File: modules/apps/asset/asset-taglib/.eslintrc.js [hunk 13]
```diff
@@ -0,0 +1,17 @@
...
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+module.exports = {
+	extends: ['liferay/react']
```
Comment (jbalsas) [inline modules/apps/asset/asset-taglib/.eslintrc.js]: Do we still need this, @wincent? Will it conflict with other presets if we enable it by default?

File: modules/apps/asset/asset-taglib/src/main/java/com/liferay/asset/taglib/servlet/taglib/AssetTagsSelectorTag.java [hunk 14]
```diff
@@ -217,46 +257,29 @@ protected String getTagNames() {
...
 
-		for (String tagName : tagNames) {
-			Map<String, String> item = new HashMap<>();
-
-			item.put("label", tagName);
-			item.put("value", tagName);
-
-			selectedItems.add(item);
-		}
+		data.put("addCallback", _getNamespace() + _addCallback);
```
Comment (jbalsas) [inline modules/apps/asset/asset-taglib/src/main/java/com/liferay/asset/taglib/servlet/taglib/AssetTagsSelectorTag.java]: Is this correct? What happens if we don't pass a callback? I assume we will still try to call something because this will be truthy because of the `namespace` addition.

We could optimize a little bit by checking if `Validator.isNotNull(_addCallback)` and only then setting it.

File: modules/apps/asset/asset-taglib/src/main/java/com/liferay/asset/taglib/servlet/taglib/AssetTagsSelectorTag.java [hunk 15]
```diff
@@ -217,46 +257,29 @@ protected String getTagNames() {
...
+		data.put("eventName", getEventName());
+		data.put("groupIds", getGroupIds());
+		data.put("inputName", _getInputName());
+		data.put("portletURL", getPortletURL().toString());
+		data.put("removeCallback", _getNamespace() + _removeCallback);
```
Comment (jbalsas) [inline modules/apps/asset/asset-taglib/src/main/java/com/liferay/asset/taglib/servlet/taglib/AssetTagsSelectorTag.java]: Similar to the `removeCallback` comment

File: modules/apps/asset/asset-taglib/src/main/java/com/liferay/asset/taglib/servlet/taglib/AssetTagsSelectorTag.java [hunk 16]
```diff
@@ -217,46 +257,29 @@ protected String getTagNames() {
...
+		data.put("showSelectButton", _showSelectButton);
+		data.put("tagNames", getTagNames());
```
Comment (jbalsas) [inline modules/apps/asset/asset-taglib/src/main/java/com/liferay/asset/taglib/servlet/taglib/AssetTagsSelectorTag.java]: Do we still need `tagNames`? Now that we don't have soy and we can do whatever manipulations we need, maybe we can just keep one, can't we?

File: modules/apps/asset/asset-taglib/.babelrc (hunk 12 above)
Comment (wincent) [reply modules/apps/asset/asset-taglib/.babelrc]: I think we should enable it by default very carefully. For example, it will turn JSX into calls to `React.createElement()`, which obviously might clash with components using "incremental-dom"/"metal-jsx". 

When should we do this? When we only had 1 or 2 places using React, it made sense to have them explicitly opt-in. At some point, it starts to make more sense to opt in to React by default and have Metal-using modules opt-out. Are we there yet? I don't know, but if we aren't, I hope to be there soon.

File: modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_tags_selector/page.jsp [hunk 17]
```diff
@@ -0,0 +1,63 @@
...
+		</c:forEach>
+	
+		<div class="form-group">
+			<label><%= LanguageUtil.get(request, "tags") %></label>
+
+			<div class="input-group input-group-stacked-sm-down">
+				<div class="input-group-item">
+					<div class="form-control form-control-tag-group">
+						<c:forEach items="<%= selectedItems %>" var="selectedItem">
+							<span tabindex="0" class="label label-dismissible label-secondary">
```
Comment (jbalsas) [inline modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_tags_selector/page.jsp]: Could we try using `clay:label` here? Might be simpler and hide some of these markup away.

If not, we could at least use `clay:button` or `clay:icon`. Just so we don't have to maintain all these in so many places.

File: modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_tags_selector/page.jsp [hunk 18]
```diff
@@ -0,0 +1,63 @@
...
+
+<%
+Map<String, Object> data = (Map<String, Object>)request.getAttribute("liferay-asset:asset-tags-selector:data");
+List<String> selectedItems = (List<String>)data.get("selectedItems");
+String inputName = (String)data.get("inputName");
+%>
+
+<div>
+	<div class="lfr-tags-selector-content">
+		<c:forEach items="<%= selectedItems %>" var="selectedItem">
```
Comment (jbalsas) [inline modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_tags_selector/page.jsp]: I don't think we use `c:forEach` in jsps? Common pattern would be:

```
<%
    for (String selectedItem : selectedItems) {
%>
    <input name="<%= inputName %>" type="hidden" value="<%= selectedItem %>">
<%
}
%>

File: modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_tags_selector/page.jsp [hunk 19]
```diff
@@ -0,0 +1,63 @@
...
+			<input name="<%= inputName %>" type="hidden" value="${selectedItem}">
+		</c:forEach>
+	
+		<div class="form-group">
+			<label><%= LanguageUtil.get(request, "tags") %></label>
+
+			<div class="input-group input-group-stacked-sm-down">
+				<div class="input-group-item">
+					<div class="form-control form-control-tag-group">
+						<c:forEach items="<%= selectedItems %>" var="selectedItem">
```
Comment (jbalsas) [inline modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_tags_selector/page.jsp]: Same here, no `c:forEach` as far as I know.

File: modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_tags_selector/page.jsp [hunk 20]
```diff
@@ -0,0 +1,63 @@
...
+							</span>
+						</c:forEach>
+
+						<input class="form-control-inset" type="text" value="">
+					</div>
+				</div>
+			</div>
+		</div>
+
+		<button class="btn btn-secondary" type="button"><%= LanguageUtil.get(request, "select") %></button>
```
Comment (jbalsas) [inline modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_tags_selector/page.jsp]: Not sure if this or `<liferay-ui:message key="select" />` is preferred...

File: modules/apps/asset/asset-taglib/.eslintrc.js (hunk 13 above)
Comment (wincent) [reply modules/apps/asset/asset-taglib/.eslintrc.js]: Similar story here. The two presets to worry about are [the React one](https://github.com/liferay/eslint-config-liferay/blob/master/react.js) and [the Metal one](https://github.com/liferay/eslint-config-liferay/blob/master/metal.js).

If we turn the React one on everywhere by default, we'll start getting errors in Metal components: we need to make sure that if they opt-in to the "metal" preset, then it suppresses the "react" one rather than just augmenting it.

In short, will have to try it out carefully and make adjustments if necessary.

File: modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_tags_selector/soy/AssetTagsSelector.es.js [hunk 21]
```diff
@@ -0,0 +1,328 @@
...
+	 * @review
+	 * @type {?string}
+	 */
+	tagNames: Config.string().value('')
+};
+
+Soy.register(AssetTagsSelector, templates);
+
+export {AssetTagsSelector};
+export default AssetTagsSelector;
```
Comment (wincent) [inline modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_tags_selector/soy/AssetTagsSelector.es.js]: I think we should stop doing these double exports. I'd just do the default export.

File: modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_tags_selector/AssetTagsSelector.es.js [hunk 22]
```diff
@@ -12,118 +12,86 @@
...
+			window.location.origin +
+			themeDisplay.getPathContext() +
+			'/api/jsonws/invoke'
+	});
+
+	useEffect(() => {
+		if (onInputValueChange) {
+			onInputValueChange(innerInputValue);
+		}
+	}, [innerInputValue, onInputValueChange]);
```
Comment (wincent) [inline modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_tags_selector/AssetTagsSelector.es.js]: Just for reference, whenever you add a callback to a dependencies array, it's good to verify the conditions under which you expect it to change. For example, you might want to make sure that, if passed from a parent, the parent is using `useCallback()` to memoize the callback and not have it change every time.

I can't seen any callers in this PR, so I can't evaluate that, but I do have one concern here which is that if it every changes, then we *will* call it immediately, even if the value doesn't change. That sounds like it might be unintended. eg. this sequence of events:

1. Value changes.
2. We call the callback.
3. Callback changes, but not the value.
4. We call the new callback even though the value didn't change.

Now, this might be harmless because perhaps the callback never changes, but it is something to think about. I think what you really intend here is not "call callback whenever it or the value changes" but rather "whenever the value changes, call the callback (if we have one)". So if that is your intent, you would benefit from the `usePrevious` hook and you could right this as:

```javascript
const previousInnerInputValue = usePrevious(innerInputValue);

if (innerInputValue !== previousInnerInputValue) {
    if (onInputValueChange) {
        onInputValueChange(innerInputValue);
    }
}
```

File: modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_tags_selector/AssetTagsSelector.es.js [hunk 23]
```diff
@@ -12,118 +12,86 @@
...
 
-				this.tagNames = this._getTagNames();
+	useEffect(() => {
+		if (onSelectedItemsChange) {
+			onSelectedItemsChange(innerSelectedItems);
+		}
+	}, [innerSelectedItems, onSelectedItemsChange]);
```
Comment (wincent) [inline modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_tags_selector/AssetTagsSelector.es.js]: Same concern here with what will happen if the callback changes.

File: modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_tags_selector/AssetTagsSelector.es.js (hunk 9 above)
Comment (wincent) [reply modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_tags_selector/AssetTagsSelector.es.js]: So, if `innerInputValue` changes from not-falsey, to falsey, we won't refetch. Is that the expected behavior here? I suspect it is and we'll update the UI by rendering `<ClayInputWithMultiSelect />` with the falsey `inputValue`, but I just wanted to check.

File: modules/apps/asset/asset-taglib/src/main/java/com/liferay/asset/taglib/servlet/taglib/AssetTagsSelectorTag.java (hunk 14 above)
Comment (carloslancha) [reply modules/apps/asset/asset-taglib/src/main/java/com/liferay/asset/taglib/servlet/taglib/AssetTagsSelectorTag.java]: @jbalsas ok, we tried to not change too much what was there before, that's why we didn't change that kind of things, but I'll add it ;)

File: modules/apps/asset/asset-taglib/src/main/java/com/liferay/asset/taglib/servlet/taglib/AssetTagsSelectorTag.java (hunk 16 above)
Comment (carloslancha) [reply modules/apps/asset/asset-taglib/src/main/java/com/liferay/asset/taglib/servlet/taglib/AssetTagsSelectorTag.java]: Yep, I still need to find out if old input with selected items values in the format `one,two,three` is needed somewhere, but we can get that from the `selectedItems` list

File: modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_tags_selector/page.jsp (hunk 18 above)
Comment (carloslancha) [reply modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_tags_selector/page.jsp]: Ok, I saw `c:if` in a quick search in portal. Changing it.

**Comment by julien** [comment]
Just started reviewing :)

File: modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_tags_selector/AssetTagsSelector.es.js (hunk 9 above)
Comment (carloslancha) [reply modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_tags_selector/AssetTagsSelector.es.js]: Yep

File: modules/apps/asset/asset-taglib/src/main/java/com/liferay/asset/taglib/servlet/taglib/AssetTagsSelectorTag.java (hunk 14 above)
Comment (carloslancha) [reply modules/apps/asset/asset-taglib/src/main/java/com/liferay/asset/taglib/servlet/taglib/AssetTagsSelectorTag.java]: Done

File: modules/apps/asset/asset-taglib/src/main/java/com/liferay/asset/taglib/servlet/taglib/AssetTagsSelectorTag.java (hunk 15 above)
Comment (carloslancha) [reply modules/apps/asset/asset-taglib/src/main/java/com/liferay/asset/taglib/servlet/taglib/AssetTagsSelectorTag.java]: Done

File: modules/apps/asset/asset-taglib/src/main/java/com/liferay/asset/taglib/servlet/taglib/AssetTagsSelectorTag.java (hunk 16 above)
Comment (carloslancha) [reply modules/apps/asset/asset-taglib/src/main/java/com/liferay/asset/taglib/servlet/taglib/AssetTagsSelectorTag.java]: Done

File: modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_tags_selector/page.jsp (hunk 17 above)
Comment (carloslancha) [reply modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_tags_selector/page.jsp]: Done

File: modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_tags_selector/page.jsp (hunk 18 above)
Comment (carloslancha) [reply modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_tags_selector/page.jsp]: Done

File: modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_tags_selector/page.jsp (hunk 19 above)
Comment (carloslancha) [reply modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_tags_selector/page.jsp]: Done

File: modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_tags_selector/page.jsp (hunk 20 above)
Comment (carloslancha) [reply modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_tags_selector/page.jsp]: Done

File: modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_tags_selector/AssetTagsSelector.es.js (hunk 22 above)
Comment (carloslancha) [reply modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_tags_selector/AssetTagsSelector.es.js]: Done

File: modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_tags_selector/AssetTagsSelector.es.js (hunk 23 above)
Comment (carloslancha) [reply modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_tags_selector/AssetTagsSelector.es.js]: Done

File: modules/apps/asset/asset-taglib/src/main/java/com/liferay/asset/taglib/servlet/taglib/AssetTagsSelectorTag.java [hunk 24]
```diff
@@ -217,46 +257,35 @@ protected String getTagNames() {
...
 
-		context.put("selectedItems", selectedItems);
+		data.put("showSelectButton", _showSelectButton);
+
+		List<String> selectedItems = StringUtil.split(getTagNames());
 
-		context.put(
-			"spritemap",
-			themeDisplay.getPathThemeImages() + "/lexicon/icons.svg");
+		data.put("selectedItems", selectedItems);
```
Comment (jbalsas) [inline modules/apps/asset/asset-taglib/src/main/java/com/liferay/asset/taglib/servlet/taglib/AssetTagsSelectorTag.java]: I think this is not sorted? **se**lected vs **sh**owSelectButton?

File: modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_tags_selector/page.jsp [hunk 25]
```diff
@@ -0,0 +1,69 @@
...
+<%
+Map<String, Object> data = (Map<String, Object>)request.getAttribute("liferay-asset:asset-tags-selector:data");
+List<String> selectedItems = (List<String>)data.get("selectedItems");
+String inputName = (String)data.get("inputName");
+%>
+
+<div>
+	<div class="lfr-tags-selector-content">
+		<%
+			for (String selectedItem : selectedItems) {
```
Comment (jbalsas) [inline modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_tags_selector/page.jsp]: Run SF here, this should be de-dented, I think, as shuld the closing `}`

File: modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/view.jsp (hunk 2 above)
Comment (jbalsas) [reply modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/view.jsp]: Remember to send this to 🍋so they can double-check :)

File: modules/apps/asset/asset-taglib/src/main/java/com/liferay/asset/taglib/servlet/taglib/AssetTagsSelectorTag.java (hunk 24 above)
Comment (carloslancha) [reply modules/apps/asset/asset-taglib/src/main/java/com/liferay/asset/taglib/servlet/taglib/AssetTagsSelectorTag.java]: Done

File: modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_tags_selector/page.jsp (hunk 25 above)
Comment (carloslancha) [reply modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_tags_selector/page.jsp]: Done

File: modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/categorization/tags/EditTags.es.js [hunk 26]
```diff
@@ -0,0 +1,55 @@
...
+import EditTagsModal from './EditTagsModal.es';
+
+function EditTags(props) {
+	const [fileEntires, setFileEntries] = useState();
+	const [selectAll, setSelectAll] = useState();
+	const [folderId, setFolderId] = useState();
+	const [showModal, setShowModal] = useState();
+
+	const handleOnClose = () => {
+		setShowModal(false);
```
Comment (julien) [inline modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/categorization/tags/EditTags.es.js]: I think we should unmount the modal instead of keeping it around

File: modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/categorization/tags/EditTagsModal.es.js [hunk 27]
```diff
@@ -0,0 +1,271 @@
...
+	});
+
+	useEffect(() => {
+		setBulkSelection(getBulkSelection());
+	}, [fileEntries, folderId, getBulkSelection, repositoryId, selectAll]);
```
Comment (julien) [inline modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/categorization/tags/EditTagsModal.es.js]: In theory these props aren't going to change
